app/controllers/welcome_controller.rb

```diff
@@ -106,12 +106,11 @@ def dashboard_index
       # get list of all reports for the mission, for the dropdown
       @reports = Report::Report.accessible_by(current_ability).by_name
 
-      unless fragment_exist?(@cache_key + '/report_pane')
-        prepare_report
-      end
-
       # render JSON if ajax request
       if request.xhr?
+        # Load report asynchronously
+        prepare_report
+
         data = {
           recent_responses: render_to_string(partial: 'recent_responses'),
```

Comment:
This feels a bit hackish right now.

I think a better way to do this would be:
* To make a new endpoint on ReportsController called /data or so
* Make both the dashboard and reports#show use it, so reports#show only loads via AJAX also. This would be better also since having a loading indicator on part of the page is better than having the whole page take a long time  to load.
* Then we wouldn't need to load _main.html.erb at all on non AJAX requests, so the awkward if block in ReportController can go away, right?